core/src/test/scala/org/apache/spark/scheduler/TaskResultGetterSuite.scala

```diff
@@ -81,6 +81,16 @@ class TaskResultGetterSuite extends SparkFunSuite with BeforeAndAfter with Local
   // as we can make it) so the tests don't take too long.
   def conf: SparkConf = new SparkConf().set("spark.akka.frameSize", "1")
 
+  test("Kryo serializer for TaskResult") {
+    val conf1 = new SparkConf(false)
+    conf1.set("spark.serializer", "org.apache.spark.serializer.KryoSerializer")
+    conf1.set("spark.kryoserializer.buffer", "1m")
+    conf1.set("spark.kryoserializer.buffer.max", "2m")
+    sc = new SparkContext("local", "test", conf1)
+    val result = sc.parallelize(Seq(1), 1).map(x => 2 * x).reduce((x, y) => x)
+    assert(result === 2)
```

Comment:
It's a more specific test. In case we didn't set this one up correctly for some reason this may still pass even if kryo doesn't actually work.